src/base/fs_unix.h

```diff
@@ -8,6 +8,8 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <stdio.h>
+#include <limits.h>
```

Comment:
Could you please try `<cstdio>` and `<climits>` instead of the C version of these header files (and group them below with `<cstdlib>` and `<ctime>`) to see if they work on FreeBSD?